stream-chat-android-core/src/main/java/io/getstream/chat/android/core/internal/fsm/FiniteStateMachine.kt

```diff
@@ -52,48 +53,30 @@ public class FiniteStateMachine<S : Any, E : Any>(
         }
     }
 
+    /**
+     * The current state as [StateFlow].
+     */
+    public val stateFlow: StateFlow<S> = _state
```

Comment:
Added an `StateFlow` to our `FiniteStateMachine` to be able to observe whatever changes on it using coroutines.

Reply:
Nice!